From 6a65df63459b5f08f65ba5fc7872829624cf4a38 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sat, 12 May 2018 11:24:33 +0200
Subject: [PATCH 9/9] build: Include LineageOS specific properties in
 build.prop

 * Our properties were supposed to go to /system/etc/prop.default
    after the following commit:
    "lineage: Move to Google's method of defining system default props"
    Change-Id: I6cb0e28a7599b010b389cc541015a37010a00f4b

 * However if BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED is not true,
    only /default.prop will retain the properties contents of
    ADDITIONAL_DEFAULT_PROPERTIES and PRODUCT_SYSTEM_DEFAULT_PROPERTIES,
    and none of our versioning identification was held in the system

 * Enabling BOARD_PROPERTY_OVERRIDES_SPLIT_ENABLED globally would
    break all properties on devices that handle partitions usually
    at the device level rather than the kernel due to mounting races

 * Include vendor/lineage/config/versions.mk to be allowed to
    extend ADDITION_BUILD_PROPERTIES we need to define our releases

Change-Id: I19918eece0f0dd3ee967db0339b4d09c9a6d540e
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 core/main.mk | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/main.mk b/core/main.mk
index 3d42c8f72..a1b68ce12 100644
--- a/core/main.mk
+++ b/core/main.mk
@@ -325,6 +325,10 @@ ADDITIONAL_BUILD_PROPERTIES += net.bt.name=Android
 # with SIGQUIT. Stack trace dumping is turned on for all android builds.
 ADDITIONAL_BUILD_PROPERTIES += dalvik.vm.stack-trace-dir=/data/anr
 
+# ------------------------------------------------------------
+# Include vendor specific additions to build versions properties
+-include vendor/lineage/config/versions.mk
+
 # ------------------------------------------------------------
 # Define a function that, given a list of module tags, returns
 # non-empty if that module should be installed in /system.
-- 
2.17.0

